server/get_teams_data.go

```diff
@@ -99,14 +98,14 @@ func (p *Plugin) GetMSTeamsChannelDetailsForAllTeams(msTeamsTeamIDsVsChannelsQue
 func (p *Plugin) GetMSTeamsTeamList(userID string, r *http.Request) ([]*msteams.Team, int, error) {
 	var client msteams.Client
 	var err error
-	if r.Context().Value(ContextTokenKey) == nil {
+	if r.Context().Value(ContextClientKey) == nil {
 		client, err = p.GetClientForUser(userID)
```

Comment:
If the client in the context is nil, why are we again trying to get client for user? We should check if the client is nil in the middleware only and return from there only.

Reply:
We are using the same functions for slash commands as well and they don't have middleware. We can still return from the middleware but we cant remove that condition.

Reply:
Fixed in #20